apps/src/lab2/projects/SourcesStore.ts

```diff
@@ -66,13 +66,23 @@ export class RemoteSourcesStore implements SourcesStore {
     channelId: string,
     sources: ProjectSources,
     projectType?: ProjectType,
-    replace = false
+    forceNewVersion = false
   ) {
     let options = undefined;
     if (this.currentVersionId) {
+      // If forceNewVersion is set to true, we will not replace the existing version (i.e., we will create
+      // a new version). Otherwise we check if we should replace the existing version based on the last new
+      // version saved in this session.
+      const shouldReplaceExistingVersion = forceNewVersion
+        ? false
+        : this.shouldReplaceExistingVersion();
```

Comment:
I'm finding it a little bit confusing that we have a class method and a local boolean with the same name. Could we consider renaming one or the other? Perhaps the const above could be `versionReplacementRequired` or something?

Secondly, and more subjectively, I find ternaries harder to read than a simple `&&` chain. I wonder if `!forceNewVersion && this.shouldReplaceExistingVersion()` is more readable while being equivalent. No need to change this if you prefer the current way!

Reply:
I'll rename it! 
Doing these booleans broke my brain (I think because `false` means `create a new version`). I personally found the ternary easier to understand while writing this because if `forceNewVersion` is true we don't care what `shouldReplaceExistingVersion` is. But I know others prefer chains in general, since they are equivalent I'll update it!

Reply:
> if `forceNewVersion` is true we don't care what `shouldReplaceExistingVersion` is

This is actually exactly how I interpret the chain as well. For example, with `a && b` JS never bothers to evaluate `b` if `a` is false. I think others (maybe @ebeastlake?) have disagreed with which is more intuitive. Perhaps it's worth more discussion, but certainly not in any way that should hold up this PR!

Reply:
updated to use a chain, it's definitely the more "standard" way 😄